### ISSUE TITLE
[Bug]: indeterminate state for `sp-swatch` shows an "Elements must only use supported ARIA attributes" error

### ISSUE THREAD

**Issue by Rocss** [adobe/spectrum-web-components#4518]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-swatch

### Expected behavior

No accessibility warnings should be thrown

### Actual behavior

Issue was observed in Express, but it is reproducing on Spectrum Web Components documentation as well: https://opensource.adobe.com/spectrum-web-components/components/swatch/

### Screenshots

<img width="1261" alt="Screenshot 2024-05-29 at 14 12 58" src="https://github.com/adobe/spectrum-web-components/assets/13311865/48fc138a-5372-47f4-baae-dd4f10b2f3b9">

<img width="811" alt="Screenshot 2024-05-29 at 15 03 21" src="https://github.com/adobe/spectrum-web-components/assets/13311865/27a4ed83-e588-4e8a-8af7-60a063987760">

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/swatch/
2. Open the axe-devtools extension and run a full scan of the page
3. Observe the accessibility error: "Ensures an element's role supports its ARIA attributes"

### Sample code that illustrates the problem

While `aria-checked="mixed"` would be useful in a `sp-swatch-group` where the role of the `sp-swatch` would be `checkbox`, in a single `sp-swatch` which surfaces the `role=button`, the use of aria-checked is invalid. 
The "mixed" state should be part of the aria-label in this case.
